drivers/wireless/lpwan/sx127x/sx127x.c

```diff
@@ -4589,15 +4589,15 @@ int sx127x_register(FAR struct spi_dev_s *spi,
 
   dev->pfd = NULL;
 
-  /* Initialize sem */
+  /* Initialize mutex & sem */
 
-  nxsem_init(&dev->dev_sem, 0, 1);
+  nxmutex_init(&dev->dev_lock);
```

Comment:
ditto. add vs not add `nxmutex_destroy` and `nxsem_destroy` if `register_driver` fails